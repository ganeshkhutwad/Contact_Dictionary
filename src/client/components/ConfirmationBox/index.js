export { default as ConfirmationBox } from './ConfirmationBox';