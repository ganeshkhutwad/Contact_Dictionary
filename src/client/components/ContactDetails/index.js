export { default as ContactDetails } from './ContactDetails';