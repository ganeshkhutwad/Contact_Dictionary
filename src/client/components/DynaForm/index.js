export { default as DynaForm } from './DynaForm';
