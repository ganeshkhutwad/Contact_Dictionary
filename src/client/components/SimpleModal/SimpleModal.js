/**
@author Ganesh Khutwad
 */
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Modal from '@material-ui/core/Modal';

function getModalStyle() {
  const top = 50;
  const left = 50;

  return {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
  };
}

const styles = theme => ({
  paper: {
    position: 'absolute',
    width: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing.unit * 4,
  },
});

const SimpleModal = ({classes, open, closeModal, children}) => {
    return (
        <Modal
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          open={open}
          onClose={closeModal}
        >
          <div style={getModalStyle()} className={classes.paper}>
            {children}
          </div>
        </Modal>
    )
}

SimpleModal.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(SimpleModal)