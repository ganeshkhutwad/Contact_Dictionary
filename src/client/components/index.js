export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
