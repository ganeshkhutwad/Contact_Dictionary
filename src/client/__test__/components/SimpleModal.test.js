/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { SimpleModal } from '../../components/SimpleModal';

const classes = { };
const open = true;
const closeModal = () => { };

describe('`SimpleModal` Tests', () => {
    it('`SimpleModal` Snapshot', () => {
        const simpleModalWrapper = shallow(<SimpleModal
                                                classes={classes}
                                                open={open}
                                                closeModal={closeModal} />);
        expect(simpleModalWrapper).toMatchSnapshot();
    });
});
