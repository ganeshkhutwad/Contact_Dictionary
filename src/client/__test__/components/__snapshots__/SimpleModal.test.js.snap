// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`SimpleModal\` Tests \`SimpleModal\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SimpleModal)
    classes={Object {}}
    closeModal={[Function]}
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "paper": "SimpleModal-paper-1",
      },
      "closeModal": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "paper": "SimpleModal-paper-1",
        },
        "closeModal": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
